packages/shared/lib/utils/encryption.manager.ts

```diff
@@ -272,6 +272,32 @@ class EncryptionManager {
         return decryptedDataRecords as unknown as DataRecordWithMetadata[] | RecordWrapCustomerFacingDataRecord;
     }
 
+    public decryptDataRecord(dataRecord: RawDataRecordResult): RawDataRecordResult | null {
+        if (dataRecord === null) {
+            return dataRecord;
+        }
+
+        const record = dataRecord.record;
+
+        if (!record['encryptedValue']) {
+            return dataRecord;
+        }
+
+        const { encryptedValue, iv, authTag } = record;
+
+        const decryptedString = this.decrypt(encryptedValue, iv, authTag);
+
+        const updatedRecord = {
+            ...JSON.parse(decryptedString)
+        };
+
+        if (record['_nango_metadata']) {
+            updatedRecord['_nango_metadata'] = record['_nango_metadata'];
```

Comment:
Seems like it should be this no?
```suggestion
            updatedRecord.record['_nango_metadata'] = record['_nango_metadata'];
```

Reply:
No, but I did make some improvements to the types so it is clearer with https://github.com/NangoHQ/nango/pull/1778/commits/ee85d254e59866a3a91f5c2c5d9d2088fd6dd304